packages/components/src/tools-panel/test/index.js

```diff
@@ -425,4 +432,36 @@ describe( 'ToolsPanel', () => {
 			expect( altMenuItem ).toHaveAttribute( 'aria-checked', 'false' );
 		} );
 	} );
+
+	describe( 'panel header icon toggle', () => {
+		it( 'should continue to render shown by default item after it is toggled off via menu item', async () => {
+			render(
+				<ToolsPanel { ...defaultProps }>
+					<ToolsPanelItem { ...controlProps }>
```

Comment:
The default state of `controlProps` [should have a value](https://github.com/WordPress/gutenberg/blob/8a8987558044019e8c92eec916f91eff0e88646d/packages/components/src/tools-panel/test/index.js#L21) that would mean that it's actually "activated", as this test puts it, in the menu. If the test order is randomized or focused, this test fails. 

Some options for quick fixes might be switching `controlProps` to `altControlProps` which doesn't have a value. Better still is probably a before filter to ensure a known state of props. Or, just force an undefined value in the props used within the render call.

This might also be an issue elsewhere in these tests as well. There are plans to remove hardcoded CSS classes and improve these tests in a follow-up PR. So I think we can keep this proposed change to just this test. 

What do you think?

Reply:
Thanks a lot for pulling me up this. I'll test again and define a default state somehow, as you've suggested.

> There are plans to remove hardcoded CSS classes and improve these tests in a follow-up PR.

Testability without querying the DOM was one of the reasons why this PR introduced the aria label on the icon, so hopefully that's one less thing to change later 👍